huxley/www/js/stores/AssignmentStore.js

```diff
@@ -1,32 +1,57 @@
 /**
- * Copyright (c) 2011-2015 Berkeley Model United Nations. All rights reserved.
+ * Copyright (c) 2011-2016 Berkeley Model United Nations. All rights reserved.
  * Use of this source code is governed by a BSD License (see LICENSE).
  */
 
 'use strict';
 
-var $ = require('jquery');
+var ActionConstants = require('constants/ActionConstants');
+var AssignmentActions = require('actions/AssignmentActions');
 var Dispatcher = require('dispatcher/Dispatcher');
 var ServerAPI = require('lib/ServerAPI');
 var {Store} = require('flux/utils');
 
 
-var _assignmentPromises = {};
+var _schoolsAssignments = {};
+var _assignments = {};
 
 class AssignmentStore extends Store {
-  getAssignments(schoolID, callback) {
-    if (!_assignmentPromises[schoolID]) {
-      _assignmentPromises[schoolID] = ServerAPI.getAssignments(schoolID);
+  getAssignments(schoolID) {
+    if (_schoolsAssignments[schoolID]) {
+      return _schoolsAssignments[schoolID];
     }
-    if (callback) {
-      _assignmentPromises[schoolID].then(callback);
-    }
-    return _assignmentPromises[schoolID];
+
+    ServerAPI.getAssignments(schoolID).then(value => {
+      _schoolsAssignments[schoolID] = value;
+      for (const assignment of value) {
+        _assignments[assignment.id] = assignment;
+      }
+      AssignmentActions.assignmentsFetched();
```

Comment:
Same with Committees and Country, the data should be passed into the action, and the value only updated in response to an action in `__onDispatch`.

Seems a little convoluted now, but it'll be clearer as we continue to iterate on the design.